test/Microsoft.AspNetCore.Blazor.Build.Test/CodeGenerationTestBase.cs

```diff
@@ -1228,6 +1229,146 @@ public void EventHandler_OnElement_WithDelegate()
 
         #endregion
 
+        #region Generics
+
+        [Fact]
+        public void ChildComponent_Generic()
+        {
+            // Arrange
+            AdditionalSyntaxTrees.Add(Parse(@"
+using Microsoft.AspNetCore.Blazor.Components;
+
+namespace Test
+{
+    public class MyComponent<TItem> : BlazorComponent
+    {
+        [Parameter] TItem Item { get; set; }
+    }
+}
+"));
+
+            // Act
+            var generated = CompileToCSharp(@"
+@addTagHelper *, TestAssembly
+<MyComponent TItem=string Item=""@(""hi"")""/>");
```

Comment:
Good to see the syntax is `TItem=string` rather than `TItem=typeof(string)`. I'll be interested to see what level of IDE support we have for this - does it autocomplete on type names?

Reply:
yes, we put the cursor inside a `typeof( )` so you get the completion you'd get in that context. I think this is the correct thing to do because you'll get immediate feedback if you try to use a variable instead of a type constant.